### PR TITLE
feat(date-picker): add selectorButtonPlacement property

### PR DESCRIPTION
**Pull request by ryo-manba**
Closes https://github.com/nextui-org/nextui/issues/3015

## 📝 Description

Add `selectorButtonPlacement` property to `DatePicker` and `DateRangePicker`.

DatePicker with selectorButtonPlacement

## ⛳️ Current behavior (updates)

No support for customizing the calendar button position.

## 🚀 New behavior

Supports `selectorButtonPlacement` to customize the calendar button position.

## 💣 Is this a breaking change (Yes/No):

No.

## Summary by CodeRabbit

- **New Features**
  - Added support for `selectorButtonPlacement` property in DatePicker and DateRangePicker components.
  - Users can now choose the placement of the selector button (start or end) in date picker components.
  - Updated documentation to include information on the new `selectorButtonPlacement` property.

- **Documentation**
  - Enhanced documentation with a new section on Selector Button Placement for date picker components.

- **Tests**
  - Introduced new test cases to verify the functionality of `selectorButtonPlacement` and related properties.

### CODE REVIEW

File: .changeset/neat-donkeys-accept.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@nextui-org/date-picker": patch
+"@nextui-org/theme": patch
```
[the remark this answers is not in the document]
Comment (wingkwong) [reply .changeset/neat-donkeys-accept.md:3]: agree. let's target to v2.5.0.

Comment (ryo-manba) [inline .changeset/neat-donkeys-accept.md:3]:
```suggestion
"@nextui-org/date-picker": minor
"@nextui-org/theme": minor
```

**Review by jrgarciadev — APPROVED** [review]
Well done! @ryo-manba thanks!

**Comment by jrgarciadev** [comment]
Leaving this approved for the next version v2.5.0

**Pull request merged**